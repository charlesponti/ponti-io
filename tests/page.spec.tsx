import { screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import App from '../app/corona/page';
import renderWithProviders from './testUtils';

describe('App', () => {
  test('renders without crashing', () => {
    const { container } = renderWithProviders(<App />);
    expect(container).toMatchSnapshot();
  });

  // Test that the CountryPicker is rendered after loading.
  test('renders CountryPicker after loading', async () => {
    renderWithProviders(<App />);

    // Wait for the loading indicator to disappear.
    const picker = await screen.findByTestId('country-picker');
    expect(picker).toMatchSnapshot();
  });
});
