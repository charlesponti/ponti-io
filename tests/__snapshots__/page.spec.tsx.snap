// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders CountryPicker after loading 1`] = `
<label
  class="sc-gKkibJ cweQQB"
  data-testid="country-picker"
  for="country"
>
  Country
  <select
    class="_select_eda643"
    id="country"
  >
    <option
      value="global"
    >
      All
    </option>
    <option>
      US
    </option>
    <option>
      Canada
    </option>
    <option>
      China
    </option>
    <option>
      Italy
    </option>
  </select>
</label>
`;

exports[`App > renders without crashing 1`] = `
<div>
  <div
    class="flex items-center justify-center flex-col"
  >
    <div
      class="flex items-center w-full text-2xl p-4 mb-8 opacity-80"
    >
      <img
        alt="COVID 19"
        class="w-6 h-6 max-h-6 mr-2"
        data-nimg="1"
        data-testid="covid-image"
        decoding="async"
        height="24"
        loading="lazy"
        src="/_next/image?url=%2Fcovid%400.25x.webp&w=48&q=75"
        srcset="/_next/image?url=%2Fcovid%400.25x.webp&w=32&q=75 1x, /_next/image?url=%2Fcovid%400.25x.webp&w=48&q=75 2x"
        style="color: transparent;"
        width="24"
      />
      COVID-19 Data Tracker
    </div>
    <div>
      Loading...
    </div>
    <div
      class="fixed bottom-0 flex justify-center w-full text-md mt-8 mb-8"
    >
      Produced by
      <a
        class="text-md font-medium letter-spacing-1 ml-5 underline-offset-4"
        href="https://ponti.io"
        rel="noopener noreferrer"
        target="_blank"
      >
        Ponti Studios
      </a>
    </div>
  </div>
</div>
`;
